# Perfilado de datasets mineria
- Juan Pablo Mosquera
- David Zahid Riveros
- Oscar Manuel Contreras Gacha
- Anderson David Arenas Gutierrez

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

pd.set_option("display.width", 110)
rng = np.random.default_rng(2026)
print("pandas", pd.__version__)

#FIFA

In [2]:
df = pd.read_csv('fifa_eda_stats.csv')

In [3]:
"""Limpieza de valores de dinero"""
def convert_value_to_numeric(value_str):
    if pd.isna(value_str) or not isinstance(value_str, str):
        return None
    value_str = value_str.replace('€', '').strip()
    if 'M' in value_str:
        return float(value_str.replace('M', '')) * 1_000_000
    elif 'K' in value_str:
        return float(value_str.replace('K', '')) * 1_000
    elif value_str == '0':
        return 0.0
    else:
        try:
            return float(value_str)
        except ValueError:
            return None
df['Value'] = df['Value'].apply(convert_value_to_numeric)
df['Wage'] = df['Wage'].apply(convert_value_to_numeric)
df['Release Clause'] = df['Release Clause'].apply(convert_value_to_numeric)

In [4]:
print(df["Value"].value_counts(normalize=True).round(3))

Value
1100000.0     0.024
375000.0      0.020
425000.0      0.019
325000.0      0.019
450000.0      0.019
              ...  
63000000.0    0.000
51500000.0    0.000
53500000.0    0.000
58000000.0    0.000
81000000.0    0.000
Name: proportion, Length: 217, dtype: float64


In [5]:
def perfilar(df):
    filas = []
    for col in df.columns:
        s = df[col]
        fila = {
            "columna":   col,
            "tipo":      str(s.dtype),
            "distintos": s.nunique(),
            "faltan_%":  round(s.isna().mean() * 100, 1),
            "moda":      s.mode().iloc[0] if not s.mode().empty else None,
        }
        if pd.api.types.is_numeric_dtype(s):
            fila["min"]     = round(s.min(), 2)
            fila["mediana"] = round(s.median(), 2)
            fila["media"]   = round(s.mean(), 2)
            fila["max"]     = round(s.max(), 2)
        filas.append(fila)
    return pd.DataFrame(filas).set_index("columna")

perfilar(df)


,tipo,distintos,faltan_%,moda,min,mediana,media,max
columna,,,,,,,,
ID,int64,18207,0.0,16,16.0,221759.0,214298.34,246620.0
Name,object,17194,0.0,J. Rodríguez,NaN,NaN,NaN,NaN
Age,int64,29,0.0,21,16.0,25.0,25.12,45.0
Nationality,object,164,0.0,England,NaN,NaN,NaN,NaN
Overall,int64,48,0.0,66,46.0,66.0,66.24,94.0
Potential,int64,47,0.0,70,48.0,71.0,71.31,95.0
Club,object,651,1.3,AS Monaco,NaN,NaN,NaN,NaN
Value,float64,217,0.0,1100000.0,0.0,675000.0,2410695.89,118500000.0
Wage,float64,144,0.0,1000.0,0.0,3000.0,9731.31,565000.0


1. Corra `perfilar()` y pegue la salida.

<div id="df-812bf6ae-a487-4752-ad77-2c6b21f068fc" class="colab-df-container">
    <div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>tipo</th>
      <th>distintos</th>
      <th>faltan_%</th>
      <th>moda</th>
      <th>min</th>
      <th>mediana</th>
      <th>media</th>
      <th>max</th>
    </tr>
    <tr>
      <th>columna</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>ID</th>
      <td>int64</td>
      <td>18207</td>
      <td>0.0</td>
      <td>16</td>
      <td>16.0</td>
      <td>221759.0</td>
      <td>214298.34</td>
      <td>246620.0</td>
    </tr>
    <tr>
      <th>Name</th>
      <td>object</td>
      <td>17194</td>
      <td>0.0</td>
      <td>J. Rodríguez</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>Age</th>
      <td>int64</td>
      <td>29</td>
      <td>0.0</td>
      <td>21</td>
      <td>16.0</td>
      <td>25.0</td>
      <td>25.12</td>
      <td>45.0</td>
    </tr>
    <tr>
      <th>Nationality</th>
      <td>object</td>
      <td>164</td>
      <td>0.0</td>
      <td>England</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>Overall</th>
      <td>int64</td>
      <td>48</td>
      <td>0.0</td>
      <td>66</td>
      <td>46.0</td>
      <td>66.0</td>
      <td>66.24</td>
      <td>94.0</td>
    </tr>
    <tr>
      <th>Potential</th>
      <td>int64</td>
      <td>47</td>
      <td>0.0</td>
      <td>70</td>
      <td>48.0</td>
      <td>71.0</td>
      <td>71.31</td>
      <td>95.0</td>
    </tr>
    <tr>
      <th>Club</th>
      <td>object</td>
      <td>651</td>
      <td>1.3</td>
      <td>AS Monaco</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>Value</th>
      <td>object</td>
      <td>217</td>
      <td>0.0</td>
      <td>€1.1M</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>Wage</th>
      <td>object</td>
      <td>144</td>
      <td>0.0</td>
      <td>€1K</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>Preferred Foot</th>
      <td>object</td>
      <td>2</td>
      <td>0.3</td>
      <td>Right</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>International Reputation</th>
      <td>float64</td>
      <td>5</td>
      <td>0.3</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>1.11</td>
      <td>5.0</td>
    </tr>
    <tr>
      <th>Weak Foot</th>
      <td>float64</td>
      <td>5</td>
      <td>0.3</td>
      <td>3.0</td>
      <td>1.0</td>
      <td>3.0</td>
      <td>2.95</td>
      <td>5.0</td>
    </tr>
    <tr>
      <th>Skill Moves</th>
      <td>float64</td>
      <td>5</td>
      <td>0.3</td>
      <td>2.0</td>
      <td>1.0</td>
      <td>2.0</td>
      <td>2.36</td>
      <td>5.0</td>
    </tr>
    <tr>
      <th>Work Rate</th>
      <td>object</td>
      <td>9</td>
      <td>0.3</td>
      <td>Medium/ Medium</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>Body Type</th>
      <td>object</td>
      <td>10</td>
      <td>0.3</td>
      <td>Normal</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>Position</th>
      <td>object</td>
      <td>27</td>
      <td>0.3</td>
      <td>ST</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>Jersey Number</th>
      <td>float64</td>
      <td>99</td>
      <td>0.3</td>
      <td>8.0</td>
      <td>1.0</td>
      <td>17.0</td>
      <td>19.55</td>
      <td>99.0</td>
    </tr>
    <tr>
      <th>Joined</th>
      <td>object</td>
      <td>1736</td>
      <td>8.5</td>
      <td>Jul 1, 2018</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>Loaned From</th>
      <td>object</td>
      <td>341</td>
      <td>93.1</td>
      <td>Atalanta</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>Contract Valid Until</th>
      <td>object</td>
      <td>36</td>
      <td>1.6</td>
      <td>2019</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>Height</th>
      <td>object</td>
      <td>21</td>
      <td>0.3</td>
      <td>6'0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>Weight</th>
      <td>object</td>
      <td>57</td>
      <td>0.3</td>
      <td>165lbs</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>Crossing</th>
      <td>float64</td>
      <td>89</td>
      <td>0.3</td>
      <td>62.0</td>
      <td>5.0</td>
      <td>54.0</td>
      <td>49.73</td>
      <td>93.0</td>
    </tr>
    <tr>
      <th>Finishing</th>
      <td>float64</td>
      <td>93</td>
      <td>0.3</td>
      <td>58.0</td>
      <td>2.0</td>
      <td>49.0</td>
      <td>45.55</td>
      <td>95.0</td>
    </tr>
    <tr>
      <th>HeadingAccuracy</th>
      <td>float64</td>
      <td>91</td>
      <td>0.3</td>
      <td>58.0</td>
      <td>4.0</td>
      <td>56.0</td>
      <td>52.30</td>
      <td>94.0</td>
    </tr>
    <tr>
      <th>ShortPassing</th>
      <td>float64</td>
      <td>85</td>
      <td>0.3</td>
      <td>64.0</td>
      <td>7.0</td>
      <td>62.0</td>
      <td>58.69</td>
      <td>93.0</td>
    </tr>
    <tr>
      <th>Volleys</th>
      <td>float64</td>
      <td>87</td>
      <td>0.3</td>
      <td>49.0</td>
      <td>4.0</td>
      <td>44.0</td>
      <td>42.91</td>
      <td>90.0</td>
    </tr>
    <tr>
      <th>Dribbling</th>
      <td>float64</td>
      <td>94</td>
      <td>0.3</td>
      <td>64.0</td>
      <td>4.0</td>
      <td>61.0</td>
      <td>55.37</td>
      <td>97.0</td>
    </tr>
    <tr>
      <th>Curve</th>
      <td>float64</td>
      <td>89</td>
      <td>0.3</td>
      <td>58.0</td>
      <td>6.0</td>
      <td>48.0</td>
      <td>47.17</td>
      <td>94.0</td>
    </tr>
    <tr>
      <th>FKAccuracy</th>
      <td>float64</td>
      <td>90</td>
      <td>0.3</td>
      <td>42.0</td>
      <td>3.0</td>
      <td>41.0</td>
      <td>42.86</td>
      <td>94.0</td>
    </tr>
    <tr>
      <th>LongPassing</th>
      <td>float64</td>
      <td>84</td>
      <td>0.3</td>
      <td>62.0</td>
      <td>9.0</td>
      <td>56.0</td>
      <td>52.71</td>
      <td>93.0</td>
    </tr>
    <tr>
      <th>BallControl</th>
      <td>float64</td>
      <td>90</td>
      <td>0.3</td>
      <td>65.0</td>
      <td>5.0</td>
      <td>63.0</td>
      <td>58.37</td>
      <td>96.0</td>
    </tr>
    <tr>
      <th>Acceleration</th>
      <td>float64</td>
      <td>86</td>
      <td>0.3</td>
      <td>69.0</td>
      <td>12.0</td>
      <td>67.0</td>
      <td>64.61</td>
      <td>97.0</td>
    </tr>
    <tr>
      <th>SprintSpeed</th>
      <td>float64</td>
      <td>85</td>
      <td>0.3</td>
      <td>68.0</td>
      <td>12.0</td>
      <td>67.0</td>
      <td>64.73</td>
      <td>96.0</td>
    </tr>
    <tr>
      <th>Agility</th>
      <td>float64</td>
      <td>81</td>
      <td>0.3</td>
      <td>68.0</td>
      <td>14.0</td>
      <td>66.0</td>
      <td>63.50</td>
      <td>96.0</td>
    </tr>
    <tr>
      <th>Reactions</th>
      <td>float64</td>
      <td>68</td>
      <td>0.3</td>
      <td>65.0</td>
      <td>21.0</td>
      <td>62.0</td>
      <td>61.84</td>
      <td>96.0</td>
    </tr>
    <tr>
      <th>Balance</th>
      <td>float64</td>
      <td>81</td>
      <td>0.3</td>
      <td>68.0</td>
      <td>16.0</td>
      <td>66.0</td>
      <td>63.97</td>
      <td>96.0</td>
    </tr>
    <tr>
      <th>ShotPower</th>
      <td>float64</td>
      <td>92</td>
      <td>0.3</td>
      <td>68.0</td>
      <td>2.0</td>
      <td>59.0</td>
      <td>55.46</td>
      <td>95.0</td>
    </tr>
    <tr>
      <th>Jumping</th>
      <td>float64</td>
      <td>74</td>
      <td>0.3</td>
      <td>70.0</td>
      <td>15.0</td>
      <td>66.0</td>
      <td>65.09</td>
      <td>95.0</td>
    </tr>
    <tr>
      <th>Stamina</th>
      <td>float64</td>
      <td>85</td>
      <td>0.3</td>
      <td>68.0</td>
      <td>12.0</td>
      <td>66.0</td>
      <td>63.22</td>
      <td>96.0</td>
    </tr>
    <tr>
      <th>Strength</th>
      <td>float64</td>
      <td>74</td>
      <td>0.3</td>
      <td>68.0</td>
      <td>17.0</td>
      <td>67.0</td>
      <td>65.31</td>
      <td>97.0</td>
    </tr>
    <tr>
      <th>LongShots</th>
      <td>float64</td>
      <td>92</td>
      <td>0.3</td>
      <td>59.0</td>
      <td>3.0</td>
      <td>51.0</td>
      <td>47.11</td>
      <td>94.0</td>
    </tr>
    <tr>
      <th>Aggression</th>
      <td>float64</td>
      <td>85</td>
      <td>0.3</td>
      <td>65.0</td>
      <td>11.0</td>
      <td>59.0</td>
      <td>55.87</td>
      <td>95.0</td>
    </tr>
    <tr>
      <th>Interceptions</th>
      <td>float64</td>
      <td>89</td>
      <td>0.3</td>
      <td>62.0</td>
      <td>3.0</td>
      <td>52.0</td>
      <td>46.70</td>
      <td>92.0</td>
    </tr>
    <tr>
      <th>Positioning</th>
      <td>float64</td>
      <td>94</td>
      <td>0.3</td>
      <td>58.0</td>
      <td>2.0</td>
      <td>55.0</td>
      <td>49.96</td>
      <td>95.0</td>
    </tr>
    <tr>
      <th>Vision</th>
      <td>float64</td>
      <td>85</td>
      <td>0.3</td>
      <td>58.0</td>
      <td>10.0</td>
      <td>55.0</td>
      <td>53.40</td>
      <td>94.0</td>
    </tr>
    <tr>
      <th>Penalties</th>
      <td>float64</td>
      <td>87</td>
      <td>0.3</td>
      <td>45.0</td>
      <td>5.0</td>
      <td>49.0</td>
      <td>48.55</td>
      <td>92.0</td>
    </tr>
    <tr>
      <th>Composure</th>
      <td>float64</td>
      <td>84</td>
      <td>0.3</td>
      <td>60.0</td>
      <td>3.0</td>
      <td>60.0</td>
      <td>58.65</td>
      <td>96.0</td>
    </tr>
    <tr>
      <th>Marking</th>
      <td>float64</td>
      <td>92</td>
      <td>0.3</td>
      <td>60.0</td>
      <td>3.0</td>
      <td>53.0</td>
      <td>47.28</td>
      <td>94.0</td>
    </tr>
    <tr>
      <th>StandingTackle</th>
      <td>float64</td>
      <td>90</td>
      <td>0.3</td>
      <td>66.0</td>
      <td>2.0</td>
      <td>55.0</td>
      <td>47.70</td>
      <td>93.0</td>
    </tr>
    <tr>
      <th>SlidingTackle</th>
      <td>float64</td>
      <td>88</td>
      <td>0.3</td>
      <td>62.0</td>
      <td>3.0</td>
      <td>52.0</td>
      <td>45.66</td>
      <td>91.0</td>
    </tr>
    <tr>
      <th>GKDiving</th>
      <td>float64</td>
      <td>71</td>
      <td>0.3</td>
      <td>8.0</td>
      <td>1.0</td>
      <td>11.0</td>
      <td>16.62</td>
      <td>90.0</td>
    </tr>
    <tr>
      <th>GKHandling</th>
      <td>float64</td>
      <td>70</td>
      <td>0.3</td>
      <td>10.0</td>
      <td>1.0</td>
      <td>11.0</td>
      <td>16.39</td>
      <td>92.0</td>
    </tr>
    <tr>
      <th>GKKicking</th>
      <td>float64</td>
      <td>79</td>
      <td>0.3</td>
      <td>12.0</td>
      <td>1.0</td>
      <td>11.0</td>
      <td>16.23</td>
      <td>91.0</td>
    </tr>
    <tr>
      <th>GKPositioning</th>
      <td>float64</td>
      <td>79</td>
      <td>0.3</td>
      <td>10.0</td>
      <td>1.0</td>
      <td>11.0</td>
      <td>16.39</td>
      <td>90.0</td>
    </tr>
    <tr>
      <th>GKReflexes</th>
      <td>float64</td>
      <td>76</td>
      <td>0.3</td>
      <td>10.0</td>
      <td>1.0</td>
      <td>11.0</td>
      <td>16.71</td>
      <td>94.0</td>
    </tr>
    <tr>
      <th>Release Clause</th>
      <td>object</td>
      <td>1244</td>
      <td>8.6</td>
      <td>€1.1M</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
  </tbody>
</table>
</div>
    <div class="colab-df-buttons">
      
  <div class="colab-df-container">
    <button class="colab-df-convert" onclick="convertToInteractive('df-812bf6ae-a487-4752-ad77-2c6b21f068fc')"
            title="Convert this dataframe to an interactive table."
            style="display:none;">
      
  <svg xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 -960 960 960">
    <path d="M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z"/>
  </svg>
    </button>
    
  <style>
    .colab-df-container {
      display:flex;
      gap: 12px;
    }

    .colab-df-convert {
      background-color: #E8F0FE;
      border: none;
      border-radius: 50%;
      cursor: pointer;
      display: none;
      fill: #1967D2;
      height: 32px;
      padding: 0 0 0 0;
      width: 32px;
    }

    .colab-df-convert:hover {
      background-color: #E2EBFA;
      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);
      fill: #174EA6;
    }

    .colab-df-buttons div {
      margin-bottom: 4px;
    }

    [theme=dark] .colab-df-convert {
      background-color: #3B4455;
      fill: #D2E3FC;
    }

    [theme=dark] .colab-df-convert:hover {
      background-color: #434B5C;
      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);
      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));
      fill: #FFFFFF;
    }
  </style>
  </div>
  
  </div>

### Tres problemas encontrados y su solución

* **`Loaned From` con 93.1% de datos nulos:**  
  La gran mayoría de jugadores no están cedidos a préstamo, lo que genera una columna desierta de casi solo faltantes.  
  * **Solución:** Eliminar la columna por falta de varianza o convertirla en una bandera binaria `is_loaned` (1 si tiene club de origen, 0 si es `NaN`).

* **Valores numéricos guardados como texto (`Height` y `Weight`):**  
  Tienen formato de cadena tipo `6'0` y `165lbs`, lo que impide su análisis estadístico y cálculo de correlación directa.  
  * **Solución:** Aplicar una función de transformación en Pandas para convertir `Height` a centímetros y `Weight` a kilogramos (o libras numéricas) y castearlas a tipo `float64`.

* **Nulos masivos en cláusula de rescisión (`Release Clause` con 8.6% de faltantes):**  
  Existen 1,564 jugadores sin cláusula asignada en la base de datos.  
  * **Solución:** Imputar los faltantes calculando una estimación basada en la relación proporcional entre `Value` y `Overall`, o asignar 0 si el contrato no contempla dicha cláusula.

---

### Columna objetivo y su reparto (`Value`)

Al ser **`Value`** una variable continua económica (target de regresión), el método `value_counts(normalize=True)` se utiliza para observar la concentración del mercado o agrupar por rangos. Su reparto estadístico directo muestra:

* **Mínimo:** €0.0
* **Mediana:** €675,000.0
* **Media:** €2,410,695.89
* **Máximo:** €118,500,000.0

*(El reparto de `Value` presenta un marcado sesgo positivo a la derecha [right-skewed], donde más del 50% de los jugadores tiene un valor de mercado inferior a €675,000, mientras que una élite de superestrellas eleva la media a €2.4M y alcanza un pico de €118.5M).*

---

### ¿Este archivo aguanta el semestre?

**Sí, aguanta perfectamente el semestre**, porque cuenta con un volumen masivo de datos (18,207 registros), más de 80 variables con gran diversidad de tipos (métricas físicas, habilidades técnicas de 0 a 100, salarios y valor comercial) y múltiples desafíos de limpieza (nulos masivos, parseo de cadenas y transformaciones financieras) suficientes para todas las entregas de minería.